src/Components/DnDLayout/GridLayout.tsx

```diff
@@ -45,6 +46,12 @@ const getResizeHandle = (resizeHandleAxis: string, ref: React.Ref<HTMLDivElement
 };
 
 const LayoutEmptyState = () => {
+  const setDrawerExpanded = useSetAtom(drawerExpandedAtom);
+
+  useEffect(() => {
+    setDrawerExpanded(true);
```

Comment:
Rather than effect, I would add the `setDrawerExpanded(true)` to the handle change callback if the empty array condition is met.

Usually, you want to adjust your state based on an event, not an effect. Its just a better practice.

Reply:
Pushed a change here https://github.com/RedHatInsights/widget-layout/pull/55/commits/bd2f13471f13a8c2caef02473e653d0d63ed9d82, but now running into problem where on the initial run of `onLayoutChange`, the page is not yet loaded so the layout is seen as empty even if there are widgets. However, if I check also that `isLoaded`, then it skips that first check entirely.

Reply:
Added back in the effect. It looks to be working as expected now.

Where were you suggesting using `isLoaded` here? It doesn't appear like we need to add any additional checks.